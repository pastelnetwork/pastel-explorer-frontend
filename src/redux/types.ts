import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { Action } from 'redux';
import { RootState } from '@redux/store';

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export type AppThunkDispatch = ThunkDispatch<RootState, unknown, Action<string>>;
