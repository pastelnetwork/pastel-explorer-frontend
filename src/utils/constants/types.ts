import { LabelKeyObject } from 'react-csv/components/CommonPropTypes';
import { PeriodTypes, TGranularity } from '@utils/helpers/statisticsLib';
import { TStatisticsInfo } from '@utils/types/IStatistics';

export type TThemeColor = {
  name: string;
  backgroundColor: string;
  lineColor?: string;
  splitLineColor: string;
  color: string;
  stack?: string;
  smooth?: boolean;
};

export type TThemeInitOption = {
  theme?: TThemeColor | null;
  data?: number[][];
  dataX?: string[];
  dataY?: number[];
  dataY1?: number[];
  dataY2?: number[];
  chartName: string;
  minY: number;
  maxY: number;
};

export type TLineChartProps = {
  chartName: string;
  period?: PeriodTypes;
  dataX?: string[];
  dataY?: number[];
  dataY1?: number[];
  dataY2?: number[];
  title?: string;
  granularity?: TGranularity;
  info: TStatisticsInfo;
  granularities?: TGranularity[];
  offset: number;
  periods?: PeriodTypes[];
  handleGranularityFilterChange?: (_granularity: TGranularity) => void;
  handleBgColorChange: (_color: string) => void;
  handlePeriodFilterChange?: (_period: PeriodTypes) => void;
};

export type TCsvHeaderType = {
  [key: string]: LabelKeyObject[];
};

export type TScatterChartProps = {
  chartName: string;
  data: number[][];
  dataX: string[];
  title?: string;
  period?: PeriodTypes;
  info: TStatisticsInfo;
  offset: number;
  periods: PeriodTypes[];
  handleBgColorChange: (_color: string) => void;
  handlePeriodFilterChange?: (_period: PeriodTypes) => void;
  handleGranularityFilterChange?: (_granularity: TGranularity) => void;
};
