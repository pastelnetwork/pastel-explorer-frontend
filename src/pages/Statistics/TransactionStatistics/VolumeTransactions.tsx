import { useState } from 'react';
import { Skeleton } from '@material-ui/lab';

import { TLineChartData, IHashRateResponse } from '@utils/types/IStatistics';

import { PeriodTypes, transformChartData } from '@utils/helpers/statisticsLib';
import * as URLS from '@utils/constants/urls';
import { useDeferredData } from '@utils/helpers/useFetch/useFetch';
import { periods, info } from '@utils/constants/statistics';
import { useBackgroundChart } from '@utils/hooks';
import { EChartsLineChart } from '@pages/HistoricalStatistics/Chart/EChartsLineChart';

const CHART_HEIGHT = 386;

const NetworkStatistics: React.FC = () => {
  const [period, setPeriod] = useState(periods[1][0]);
  const [currentBgColor, handleBgColorChange] = useBackgroundChart();

  const { isLoading, data: chartData } = useDeferredData<IHashRateResponse, TLineChartData>(
    { method: 'get', url: `${URLS.VOLUME_TRANSACTION_URL}`, params: { period } },
    transformChartData,
    undefined,
    undefined,
    [period],
  );

  const handlePeriodFilterChange = (value: PeriodTypes) => {
    setPeriod(value);
  };
  if (!chartData || isLoading) {
    return <Skeleton animation="wave" variant="rect" height={CHART_HEIGHT} />;
  }
  return (
    <div style={{ flex: 1, backgroundColor: currentBgColor }}>
      <EChartsLineChart
        chartName="averageblocksize"
        dataX={chartData?.dataX}
        dataY={chartData?.dataY}
        title="Volume of transactions (PSL)"
        info={info}
        offset={1000}
        period={period}
        periods={periods[1]}
        handleBgColorChange={handleBgColorChange}
        handlePeriodFilterChange={handlePeriodFilterChange}
      />
    </div>
  );
};

export default NetworkStatistics;
