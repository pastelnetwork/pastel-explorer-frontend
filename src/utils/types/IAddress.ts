export interface IAddressData {
  amount: number;
  timestamp: number;
  transactionHash: string;
  direction: string;
}

export interface IAddress {
  address: string;
  incomingSum: number;
  outgoingSum: number;
  data: Array<IAddressData>;
}
