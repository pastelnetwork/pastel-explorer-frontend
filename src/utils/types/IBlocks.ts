export interface IBlockTransaction {
  id: string;
  recipientCount: number;
  totalAmount: number;
}

export interface IBlock {
  confirmations: number;
  difficulty: string;
  height: number;
  id: string;
  merkleRoot: string;
  nextBlockHash: string;
  nonce: string;
  previousBlockHash: string;
  size: number;
  solution: string;
  timestamp: number;
  transactionCount: number;
  transactions: Array<IBlockTransaction>;
}

export interface IRawBlock {
  anchor: string;
  bits: string;
  chainwork: string;
  confirmations: number;
  difficulty: string;
  finalsaplingroot: string;
  hash: string;
  height: number;
  merkleroot: string;
  nonce: string;
  previousblockhash: string;
  size: number;
  solution: string;
  time: number;
  transactions: Array<IBlockTransaction>;
  tx: string[];
}
