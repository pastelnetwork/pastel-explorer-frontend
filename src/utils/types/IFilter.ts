export type TFilter = {
  name: string;
  value: string;
};
