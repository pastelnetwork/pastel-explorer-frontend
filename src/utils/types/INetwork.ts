export interface INetworkPeers {
  city: string;
  country: string;
  id: string;
  ip: string;
  latitude: number;
  longitude: number;
  nodeId: number;
  protocol: string;
  version: number;
}

export interface INetworkSupernodes {
  address: string;
  city: string;
  country: string;
  id: string;
  ip: string;
  lastPaidBlock: number;
  lastPaidTime: number;
  latitude: number;
  longitude: number;
  port: string;
  status: string;
  snPastelIdPubkey: string;
  rankAsOfBlockHeight: number;
  protocolVersion: number;
  masternodeRank: number;
  dateTimeLastSeen: number;
  activeSeconds: number;
}

export interface INetwork {
  peers: Array<INetworkPeers>;
  masternodes: Array<INetworkSupernodes>;
}
