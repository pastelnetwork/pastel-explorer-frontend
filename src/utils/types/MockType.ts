import { TFunction } from 'i18next';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type MyMockType = jest.Mock<any, any, any> &
  TFunction<['translation', ...string[]], undefined>;
