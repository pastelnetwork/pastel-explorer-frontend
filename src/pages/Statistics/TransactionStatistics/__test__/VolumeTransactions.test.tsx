import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import 'jest-styled-components';

import { MyMockType } from '@utils/types/MockType';
import i18next from '../../../../utils/helpers/i18n';
import VolumeTransactions from '../VolumeTransactions';

jest.mock('i18next-http-backend');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            // noop
          }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {
      // noop
    },
  },
}));
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
}) as MyMockType;
const mockStore = configureMockStore();
const store = mockStore({});

describe('pages/Statistics/TransactionStatistics/VolumeTransactions', () => {
  const blockElements = [
    {
      id: '245684',
      transactionCount: '3',
      height: '245684',
      size: '2.5kB',
      minutesAgo: '3 minutes',
      ticketsCount: '0',
      status: 'entered',
    },
  ];
  const wrapper = shallow(
    <Provider store={store}>
      <VolumeTransactions blockElements={blockElements} />
    </Provider>,
  );

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
