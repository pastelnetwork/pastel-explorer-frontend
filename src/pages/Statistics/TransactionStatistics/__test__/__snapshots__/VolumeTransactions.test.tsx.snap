// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Statistics/TransactionStatistics/VolumeTransactions renders correctly 1`] = `ShallowWrapper {}`;
